src/backend/input.rs

```diff
@@ -216,6 +231,9 @@ pub trait InputBackend: Sized {
     /// Get current `InputConfig`
     fn input_config(&mut self) -> &mut Self::InputConfig;
 
+    /// Called to inform the Input backend about a new focused Output for a `Seat`
+    fn set_output_metadata(&mut self, seat: &Seat, output: &Output);
```

Comment:
If we can replicate libinput's API on the events, and make the transformation an event method, that would be the best in my opinion. This would completely eliminate the need for `set_output_metadata`, which is good. (I really would like the scope of the backends to be as small as possible, we can always build more elaborate abstractions on top of them later.)

It also would directly solve the question of devices linked to an output, as the inputbackend would not have to worry about it.

I guess to make it work, TouchEvent and PointerEvent would need to become traits, with concrete types being backend-dependant?